export.py

```diff
@@ -217,7 +217,15 @@ def export_coreml(model, im, file, int8, half, prefix=colorstr('CoreML:')):
 def export_engine(model, im, file, train, half, simplify, workspace=4, verbose=False, prefix=colorstr('TensorRT:')):
     # YOLOv5 TensorRT export https://developer.nvidia.com/tensorrt
     try:
-        import tensorrt as trt  # pip install -U nvidia-tensorrt --index-url https://pypi.ngc.nvidia.com
+        assert im.device.type != 'cpu', 'export running on CPU but must be on GPU, i.e. `python export.py --device 0`'
+        try:
+            import tensorrt as trt
+        except Exception:
+            s = f"\n{prefix} tensorrt not found and is required by YOLOv5"
+            LOGGER.info(f"{s}, attempting auto-update...")
+            r = '-U nvidia-tensorrt --index-url https://pypi.ngc.nvidia.com'
```

Comment:
FYI @glenn-jocher , These wheels only works on Ubuntu 18.04+ or CentOS 7+ with Python versions 3.6 to 3.9 and CUDA 11.x. Check out the details at https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html#installing-pip. Many user use TensorRT on Windows, perhaps some explanations should be added here to prevent user annoyance when using it, such as https://github.com/ultralytics/yolov5/issues/7009#issuecomment-1068997616 .

Reply:
@zhiqwang good point! I can't programmatically check the CUDA version, but we can check OS before install. I'll make a PR.

Reply:
@zhiqwang all done! PR https://github.com/ultralytics/yolov5/pull/7549 is merged that implements OS checks prior to install. This should resolve most of the problem, and migration over time to CUDA 11 should resolve the rest of the issue. I guess Python 3.10 could be an edge case